<?xml version="1.0" ?>
<launch>
  <!-- note, using weak passwords combined with privileged containers is a really bad idea. -->
  <!-- also on first connection we need to do `ssh -oHostKeyAlgorithms='ssh-rsa' host` on every host. this should be automated  -->
  <machine name="tm1" address="torch_machine4.lavine" env-loader="/root/dop_tch/workspace/env.sh" user="root" password="ros_ros"/>
  <machine name="tm2" address="torch_machine4.poop"   env-loader="/root/dop_tch/workspace/env.sh" user="root" password="ros_ros"/>

  <!-- division by hand here! -->

  <group ns="classifiers1">
    <remap from="train_in"    to="/train/text"/>
    <remap from="train_label" to="/train/label"/>
    <remap from="test_in"     to="/test/text"/>
    <remap from="test_label"  to="/test/label"/>

 	<!-- here I will pre-start all the classifiers to be tested and then just kill them as I get the results?  -->
    <include file="$(find ros_example_torch_classifier)/launch/simple_classifier.launch" >
      <arg name="machine" value="tm1"/>
      <arg name="param2" value="a_value"/>
    </include>
    <include file="$(find ros_example_torch_classifier)/launch/simple_classifier2.launch" >
      <arg name="machine" value="tm1"/>
      <arg name="param2" value="a_value"/>
    </include>
  </group>

  <group ns="classifiers2">
    <remap from="train_in"    to="/train/text"/>
    <remap from="train_label" to="/train/label"/>
    <remap from="test_in"     to="/test/text"/>
    <remap from="test_label"  to="/test/label"/>
 
 <!-- here I will pre-start all the classifiers to be tested and then just kill them as I get the results?  -->
    <!-- <include file="$(find ros_example_torch_classifier)/launch/simple_classifier.launch" >
      <arg name="machine" value="tm2"/>
      <arg name="param2" value="a_value"/>
    </include> -->
    <!-- <include file="$(find ros_example_torch_classifier)/launch/simple_classifier2.launch" >
      <arg name="machine" value="tm2"/>
      <arg name="param2" value="a_value"/>
    </include> -->
    <!-- <include file="$(find ros_example_torch_classifier)/launch/simple_classifier2.launch" /> -->
  </group>

</launch>
